components/ratgdo/ratgdo.cpp

```diff
@@ -441,17 +541,9 @@ namespace ratgdo {
 
         set_retry(
             500, 10, [=](uint8_t r) {
-                if (*this->door_state != DoorState::UNKNOWN) { // have status
-                    if (*this->openings != 0) { // have openings
-                        return RetryResult::DONE;
-                    } else {
-                        if (r == 0) { // failed to sync probably rolling counter is wrong, notify
-                            ESP_LOGD(TAG, "Triggering sync failed actions.");
-                            this->sync_failed = true;
-                        };
-                        this->send_command(Command::GET_OPENINGS);
-                        return RetryResult::RETRY;
-                    }
+                if (*this->door_state != DoorState::UNKNOWN) { // GET_STATUS succeeded
+                    this->query_status();  //Get openings and TTC settings to initalize
```

Comment:
This does not work ok if one of the request fails. For example if `GET_OPENINGS` doesn't receive a response it doesn't retry the request like the original code did. (The opener can choose not to respond to commands if too many are sent too quickly, or the response is not received for any number of reasons).

Reply:
I was looking at it like if GET_STATUS succeeded then the rolling code was okay.  It is possible the door sends a STATUS response by chance during the sync retry period as well.  I think adding the retry mechanism to the `query_status()` to make sure gdo sends new data for each request would make it robust for initial sync and any future calls but without the increment to the rolling counter.  Does this address your comment?

Reply:
> I was looking at it like if GET_STATUS succeeded then the rolling code was okay. 

True, but the rolling code is not the only reason of failure. I've see it happen that the GET_STATUS succeeds and GET_OPENINGS does not get a response (for various reasons as I initially mentioned).

> It is possible the door sends a STATUS response by chance during the sync retry period as well. I think adding the retry mechanism to the `query_status()` to make sure gdo sends new data for each request would make it robust for initial sync and any future calls but without the increment to the rolling counter. Does this address your comment?

Not sure what you mean "without the increment to the rolling counter". Each new command needs incrementing the counter by one. Additionally we increment the counter initially in sync in case we crashed and didn't manage to persist to flash the last counter value.

The retry mechanism is to ensure we sync the initial state correctly, rolling counter mismatch being the most likely reason for failure, but not the only one.

Reply:
> Not sure what you mean "without the increment to the rolling counter". Each new command needs incrementing the counter by one. Additionally we increment the counter initially in sync in case we crashed and didn't manage to persist to flash the last counter value.

I was referring to the initial increment by 10 in sync().  query_status() was reworked in https://github.com/ESPHome-RATGDO/esphome-ratgdo/pull/43/commits/8929b32685f84ba874bb7fd1261343e434a30b50 to retry with confirmation that each of the 4 status messages returned successfully.

Reply:
If I can get working ratgdo hardware to the point I no longer care about my HomeKit bridge, I have a good MSO (200Mhz 4 analog and 32 digital channels) with SPI decode and would be willing to try and poke around/capture data from said bridge. 

Unfortunately my ratgdo seems to be doa, and I haven’t had any luck getting a response to deal with that.